### PR TITLE
Fuzzer 2.0

### PR DESCRIPTION
**Pull request by guy-ps**
- Added a custom benchmark interface
- Added an interface to run only a subset of tests
- Made some addition to the readme:
    - Documenting the new options
    - Added a link to the google collab
- Added a function to evaluate response similarity to expected response

### CODE REVIEW

File: ps_fuzz/app_config.py [hunk 1]
```diff
@@ -150,10 +194,10 @@ def update_from_args(self, args):
                     setattr(self, key, value)
             except AttributeError:
                 logger.warning(f"Attempt to set an undefined configuration property '{key}'")
-                raise
+                sys.exit(1)
```
Comment (vitaly-ps) [inline ps_fuzz/app_config.py:197]: If not strictly necessary, leave the handling on the caller, and don't exit "out of the blue" in the middle of the flow.

Comment (vitaly-ps) [reply ps_fuzz/app_config.py:197]: If there was a reason to do so no problem, just check.

File: ps_fuzz/app_config.py [hunk 2]
```diff
@@ -150,10 +194,10 @@ def update_from_args(self, args):
...
             except Exception as e:
                 logger.error(f"Error setting {key}: {e}")
-                raise
+                sys.exit(1)
```
Comment (vitaly-ps) [inline ps_fuzz/app_config.py:200]: same same.

Comment (guy-ps) [reply ps_fuzz/app_config.py]: Same

**Review by vitaly-ps — APPROVED** [review]

**Pull request merged**